packages/analyzer/src/javascript/class-scanner.ts

```diff
@@ -182,6 +186,62 @@ export class ClassScanner implements JavaScriptScanner {
     };
   }
 
+  /**
+   * Handle the pattern where a class's superclass is declared as a separate
+   * variable, usually so that mixins can be applied in a way that is compatible
+   * with the Closure compiler.
+   */
+  private collapseEphemeralSuperclasses(classes: ClassType[]): ClassType[] {
+    const classById = new Map<string, ClassType>();
+    const classBySuperClassId = new Map<string, ClassType[]>();
+
+    for (const cls of classes) {
+      if (cls.name === undefined) {
+        continue;
+      }
+      classById.set(cls.name, cls);
+
+      if (cls.superClass !== undefined) {
+        const superClassId = cls.superClass.identifier;
+        const childClasses = classBySuperClassId.get(superClassId);
+        if (childClasses === undefined) {
+          classBySuperClassId.set(superClassId, [cls]);
+        } else {
+          childClasses.push(cls);
+        }
+      }
+    }
+
+    const ephemeralSuperClasses = new Set<ClassType>();
+    for (const [superClassId, childClasses] of classBySuperClassId) {
+      const superClass = classById.get(superClassId);
+      if (superClass === undefined) {
+        continue;
+      }
+      if (superClass.privacy !== 'private') {
```

Comment:
We should also check that it's an assignment to a local variable.

Reply:
Done, and added a comment for the requirements of an ephemeral superclass.